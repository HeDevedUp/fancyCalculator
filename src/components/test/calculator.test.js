import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../calculator/calculator';

test('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('To test the functionality of calculator component', () => {
  test('should Show 27 if buttons 2,1,+ and 6 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '1' }));
    fireEvent.click(screen.getByRole('button', { name: '+' }));
    fireEvent.click(screen.getByRole('button', { name: '6' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('27')).toBeTruthy();
  });

  test('should Show 1.75 if buttons 2,.,2,5,-,. and 5 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '.' }));
    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '5' }));
    fireEvent.click(screen.getByRole('button', { name: '-' }));
    fireEvent.click(screen.getByRole('button', { name: '.' }));
    fireEvent.click(screen.getByRole('button', { name: '5' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('1.75')).toBeTruthy();
  });
});
