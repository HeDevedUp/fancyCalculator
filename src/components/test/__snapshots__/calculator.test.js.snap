// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <h1>
     
    <div
      className="screen"
    >
      0
    </div>
  </h1>
  <div
    className="buttonBox"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orangee"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orangee"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orangee"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orangee"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="equals"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
